src/components/RoomView/ResetTeams.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { ref, remove, update, get, child } from 'firebase/database';
+import { database } from '../../utils/firebase';
+import { useSelector } from 'react-redux';
+import { RootState } from '../../store';
+
+const ResetTeams = () => {
+    const { roomId } = useSelector((state: RootState) => state.player);
+    const { status } = useSelector((state: RootState) => state.game);
+    const teamOneSpymasterRef = ref(database, `rooms/${roomId}/team-1/spymaster/`);
+    const teamTwoSpymasterRef = ref(database, `rooms/${roomId}/team-2/spymaster/`);
+    const teamOneOperativesRef = ref(database, `rooms/${roomId}/team-1/operatives/`);
+    const teamTwoOperativesRef = ref(database, `rooms/${roomId}/team-2/operatives/`);
+    const playersInRoomRef = ref(database, `rooms/${roomId}/players/`);
+
+    const updateAllPlayersTeamsToNull = (playerIdsArray: any, playersInRoomRef: any) => {
```

Comment:
there doesnt seem to be an 'updateChildren' method i can use to update all of the players team ids to null, so i did so with the fxn on line 16